addons/binding/org.openhab.binding.draytonwiser/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,429 @@
+    </channel-type>
+    
+    <channel-type id="zigbeeConnected-channel" advanced="true">
+        <item-type>Switch</item-type>
+        <label>Device connected</label>
+        <description>Is the device still connected to the ZigBee network</description>
+        <state readOnly="true" ></state>
+    </channel-type>
+    
+    <channel-type id="hotWaterSetPoint-channel">
+        <item-type>Switch</item-type>
+        <label>Set Point</label>
+        <description>Hot water heating set point switch</description>
+    </channel-type>
+    
+    <channel-type id="hotWaterBoostDuration-channel">
+        <item-type>Number</item-type>
+        <label>Boost Duration</label>
+        <description>Duration to boost hot water for</description>
+        <state readOnly="false" pattern="%.1f h" min="0" max="3" step="0.5"></state>
+    </channel-type>
+    
+    <channel-type id="roomBoostDuration-channel">
+        <item-type>Number</item-type>
+        <label>Boost Duration</label>
+        <description>Duration to boost the temperature for</description>
+        <state readOnly="false" pattern="%.1f h" min="0" max="3" step="0.5"></state>
+    </channel-type>
+    
+    <channel-type id="hotWaterBoosted-channel">
+        <item-type>Switch</item-type>
+        <label>Is Boosted</label>
+        <description>Is the hot water currently being boosted</description>
+        <state readOnly="true"></state>
+    </channel-type>
+    
+    <channel-type id="roomBoosted-channel">
+        <item-type>Switch</item-type>
+        <label>Is Boosted</label>
+        <description>Is the room temperature currently being boosted</description>
+        <state readOnly="true"></state>
+    </channel-type>
+    
+    <channel-type id="hotWaterBoostRemaining-channel">
+        <item-type>Number</item-type>
+        <label>Boost remaining</label>
+        <description>How long until the boost deactivates</description>
+        <state readOnly="true" pattern="%.0f m"></state>
+    </channel-type>
+    
+    <channel-type id="roomBoostRemaining-channel">
+        <item-type>Number</item-type>
+        <label>Boost remaining</label>
+        <description>How long until the boost deactivates</description>
+        <state readOnly="true" pattern="%.0f m"></state>
+    </channel-type>
+    
+    <channel-type id="windowStateDetection-channel" advanced="true">
+        <item-type>Switch</item-type>
+        <label>Open Window Detection</label>
+        <description>Is window detection turned on</description>
+    </channel-type>
+    
+    <channel-type id="windowState-channel">
+        <item-type>Contact</item-type>
+        <label>Window State</label>
+        <description>Is the window open or closed</description>
+    </channel-type>
+    
+    <channel-type id="masterSchedule-channel" advanced="true">
+        <item-type>String</item-type>
+        <label>Room Master Schedule</label>
+        <description>The complete JSON representation of the room schedule</description>
```

Comment:
Please do not offer channels with plain JSON as input. Channels are supposed to abstract from the technical details. Please remove this one.

Reply:
Yeah, I did wonder about this one. Ideally I want to be able to read and set the schedules that are maintained by the boiler-controller. i.e. I don't want to have my heating schedule controlled by openHAB, rather use it to set "scenes" which apply a particular schedule to my boiler-controller. The only other way I see is to expose the (up to) 52 setpoints for each room for the schedule.

Reply:
Otherwise if openhab were to control the schedule, it would always operate in manual mode, with openhab changing the setpoints.

Reply:
Is there a precedent for doing this in any other bindings already?
It feels like I'd be writing my own DSL for a schedule, just so I can convert it into the Drayton Schedule DSL.

Reply:
There is no such thing as a schedule format in other bindings. However this is impossible for the user to configure via a graphical UI. So I would vote to remove this channel.

In the future this functionality could be provided via a `RuleAction` and a `RuleTemplate` of the automation engine, its on our backlog, see: https://github.com/eclipse/smarthome/projects/3#card-8298857

With this you could provide a basic JSON POJO with place holders and provide `ConfigDescription` for the placeholders in a template. A user could then fill out these place holders and the configured action would send the customized POJO to the device.